### PR TITLE
Sock merchant

### PR DESCRIPTION
**Pull request by egijasp**

### CODE REVIEW

File: sock-merchant.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export function sockMerchant(n: number, ar: number[]): number {
+  // Write your code here
```
Comment (janishorsts) [inline sock-merchant.ts]: don't leave garbage comments behind, they don't convey any meaning

File: sock-merchant.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  const filteredArr = [...new Set(ar)];
```
Comment (janishorsts) [inline sock-merchant.ts]: `unique` instead of `filteredArr`? It describes intent clearer.

File: sock-merchant.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+  let count = 0;
+
+  filteredArr.forEach((num) => {
```
Comment (janishorsts) [inline sock-merchant.ts]: It works, but why didn't you use `reduce` which is more appropriate for your use case?

**Review by kaaposc — APPROVED** [review]

File: sock-merchant.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+export function sockMerchant(ar: number[]): number {
+  const colorsCount = ar.reduce((allColors: Color, color) => {
+    if (allColors[color]) {
+      allColors[color]++;
+      return allColors;
+    }
+    allColors[color] = 1;
+    return allColors;
+  }, {});
```
Comment (janishorsts) [inline sock-merchant.ts:13]: alternative

```suggestion
  const colorsCount = ar.reduce((allColors, color) => ({
    ...allColors, 
    color: allColors[color] ? allColors[color] + 1, 1
  }), {} as Record<string, number>);
```

File: sock-merchant.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+  return Object.values(colorsCount).reduce((count, value) => {
+    return count + Math.floor(value / 2);
+  }, 0);
+}
```
Comment (janishorsts) [inline sock-merchant.ts:18]: you can reduce code boilerplate if you have a single line in your function body

```suggestion
  return Object.values(colorsCount).reduce((count, value) => count + Math.floor(value / 2), 0);
}
```

**Review by janishorsts — APPROVED** [review]

**Pull request merged**